stats.js

```diff
@@ -0,0 +1,207 @@
+        delete aggregateByCounty[county].date;
+    }
+    return {
+        aggregateByCounty,
+        aggregateByDate,
+        byDate,
+        byCounty,
+        data
+    };
+}
+
+/**
+ * Sums the testing data and reduces it a single object.
+ */
+const sumTestingData = (records, aggregateCumulatives = false) => {
+    if (!Array.isArray(records) || records.length === 0) {
+        return records;
+    }
+
+    aggregateRecord = records.reduce((acc, record) => {
+        acc.new_positives += record.new_positives;
+        acc.total_number_of_tests += record.total_number_of_tests;
+        if (aggregateCumulatives) {
+            acc.cumulative_number_of_positives += record.cumulative_number_of_positives;
+            acc.cumulative_number_of_tests += record.cumulative_number_of_tests;
+        }
+        return acc;
+    }, Object.assign({
+        new_positives: 0,
+        total_number_of_tests: 0
+    }, aggregateCumulatives ? {
+        cumulative_number_of_positives: 0,
+        cumulative_number_of_tests: 0
+    } : {}));
+    // Use Object.assign to retain original fields.
+    return Object.assign({}, records[records.length - 1], aggregateRecord);
+}
+
+exports.handler = async function () {
+    await getConfig();
+    const s3 = new AWS.S3({ region: process.env.AWS_REGION })
+    const bucket = await getAssetsBucket()
+    const {
+        aggregateByCounty,
+        aggregateByDate,
+        byDate,
+        byCounty,
+        data
+    } = await getTestingData();
+
+    const statsObject = {
+        ACL: 'private',
+        Bucket: bucket,
+        ContentType: 'application/json',
+    }
+
+    try {
+        const result = await s3.headBucket({
+            Bucket: bucket
+        }).promise();
+        await s3.putBucketPolicy({
+            Bucket: bucket,
+            Policy: JSON.stringify({
+                'Version': '2012-10-17',
+                'Statement': [
+                    {
+                        'Sid': `${bucket}-s3-allow-bucket-readwrite`,
+                        'Effect': 'Allow',
+                        'Principal': {
+                            'AWS': 'arn:aws:iam::711655675495:root'
+                        },
+                        'Action': [
+                            's3:PutObject',
+                            's3:PutObjectAcl'
+                        ],
+                        'Resource': [
+                            `arn:aws:s3:::${bucket}/*`
+                        ]
+                    }
+                ]
+            })
+        }).promise();
+    } catch (error) {
+        if (error.statusCode === 404) {
+            try {
+                await s3.createBucket({
+                    ACL: 'authenticated-read',
+                    CreateBucketConfiguration: {
+                        LocationConstraint: process.env.AWS_REGION
+                    },
+                    Bucket: bucket
+                }).promise();
+            } catch (e) {
+            }
+        }
+    }
```

Comment:
All of this should be handled by terraform, lets just assume the bucket is in place and ready for use with the correct policies

Reply:
Done